contracts/cosmwasm-vm/cw-icon-light-client/src/contract.rs

```diff
@@ -1,39 +1,264 @@
+use std::cell::RefCell;
+
+use common::icon::icon::lightclient::v1::{ClientState, ConsensusState};
+use common::icon::icon::types::v1::MerkleProofs;
 #[cfg(not(feature = "library"))]
 use cosmwasm_std::entry_point;
-use cosmwasm_std::{Binary, Deps, DepsMut, Env, MessageInfo, Response, StdResult};
+use cosmwasm_std::{to_binary, Binary, Deps, DepsMut, Env, MessageInfo, Response, StdResult};
 use cw2::set_contract_version;
+use ibc_proto::google::protobuf::Any;
 
+use crate::constants::{
+    CLIENT_STATE_HASH, CONSENSUS_STATE_HASH, HEIGHT, MEMBERSHIP, NON_MEMBERSHIP,
+};
 use crate::error::ContractError;
+use crate::light_client::IconClient;
 use crate::msg::{ExecuteMsg, InstantiateMsg, QueryMsg};
+use crate::state::{CwContext, QueryHandler};
+use crate::traits::{Config, IContext, ILightClient};
+use prost::Message;
 
 // version info for migration info
 const CONTRACT_NAME: &str = "crates.io:cw-icon-light-client";
 const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
 
 #[cfg_attr(not(feature = "library"), entry_point)]
 pub fn instantiate(
-    _deps: DepsMut,
+    deps: DepsMut,
     _env: Env,
-    _info: MessageInfo,
-    _msg: InstantiateMsg,
+    info: MessageInfo,
+    msg: InstantiateMsg,
 ) -> Result<Response, ContractError> {
-    unimplemented!()
+    let _ = set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)
+        .map_err(|_e| ContractError::FailedToInitContract)?;
+    let config = Config::new(
+        msg.src_network_id,
+        msg.network_id,
+        msg.network_type_id,
+        info.sender,
+    );
+    let context = CwContext::new(RefCell::new(deps), _env);
+    context.insert_config(&config)?;
+    Ok(Response::default())
 }
 
 #[cfg_attr(not(feature = "library"), entry_point)]
 pub fn execute(
-    _deps: DepsMut,
+    deps_mut: DepsMut,
     _env: Env,
     _info: MessageInfo,
-    _msg: ExecuteMsg,
+    msg: ExecuteMsg,
 ) -> Result<Response, ContractError> {
-    unimplemented!()
+    let context = CwContext::new(RefCell::new(deps_mut), _env);
+    let client = IconClient::new(&context);
+    match msg {
+        ExecuteMsg::CreateClient {
+            client_id,
+            client_state,
+            consensus_state,
+        } => {
+            let client_state = ClientState::decode(client_state.as_slice())
+                .map_err(|e| ContractError::DecodeError(e))?;
+            let consensus_state = ConsensusState::decode(consensus_state.as_slice())
+                .map_err(|e| ContractError::DecodeError(e))?;
+            let (state_byte, update) =
+                client.create_client(&client_id, client_state, consensus_state)?;
+
```

Comment:
for core we need to add data to response . which will be deserialized at the core side

Reply:
@shreyasbhat0  no problem on that but here i wont have client_type info. what value core expects here?